### PR TITLE
cmd/snap-confine: mask host's apparmor config

### PR DESCRIPTION
**Pull request by zyga**
Some applications have access to the host's apparmor stack. On a system
with apparmor 3.x, a snap using even core20 as base, will have apparmor
2.x userspace tools available. If such application attempts to compile
an apparmor profile, then by the virtue of shared /etc, the host's
apparmor 3.x configuration will take effect.

Unfortunately, apparmor 2.x cannot parse apparmor 3.x configuration and
such applications fail to operate correctly. This notably affects Docker
and Multipass.

As a way out mask /etc/apparmor and /etc/apparmor.d from the shared host
/etc with those from the base snap.

For the record we tried to use snap-update-ns for this, but, as we
learned, we cannot do that using snap-update-ns. The mount profile
cannot know the revision of the base snap, because at the time the mount
profile is computed, the base snap may be unlinked. Subsequently
snap-update-ns cannot fill in this gap, as it may also run at the time
the base snap is unlinked (as in, has no current symlink). In contrast,
snap-confine runs only when the base snap is linked, so it can obtain
this information for free.

Fixes: https://bugs.launchpad.net/snapd/+bug/1898038
Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: cmd/snap-confine/mount-support.c [hunk 1]
```diff
@@ -321,9 +321,9 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		// Fixes the following bugs:
 		//  - https://bugs.launchpad.net/snap-confine/+bug/1580018
 		//  - https://bugzilla.opensuse.org/show_bug.cgi?id=1028568
-		const char *dirs_from_core[] =
-		    { "/etc/alternatives", "/etc/ssl", "/etc/nsswitch.conf",
-			NULL
+		const char *dirs_from_core[] = {
+			"/etc/alternatives", "/etc/ssl", "/etc/nsswitch.conf",
+			"/etc/apparmor", "/etc/apparmor.d", NULL
```
Comment (pedronis) [inline cmd/snap-confine/mount-support.c:326]: this probably merits a comment about:

some specifc and privileged interfaces (e.g docker-support) give access to apparmor_parser from the base snap which at a mimimum needs to use matching configuration 

and a pointer to the bug?

**Review by pedronis — COMMENTED** [review]
looks reasonable, needs a bit more comments and dedicated spread testing

**Review by mvo5 — APPROVED** [review]
Thanks for this!

**Review by pedronis — APPROVED** [review]
thank you

**Comment by alexmurray** [comment]
Thanks @zyga

**Comment by anonymouse64** [comment]
For whatever reason this didn't actually make it into 2.47, and is instead in 2.48.

**Pull request merged**